lib/bolt/pal/yaml_plan/step/script.rb

```diff
@@ -20,8 +20,12 @@ def self.required_keys
           # Returns an array of arguments to pass to the step's function call
           #
           private def format_args(body)
+            args        = body['arguments'] || []
+            pwsh_params = body['pwsh_params'] || {}
+
             opts = format_options(body)
-            opts = opts.merge('arguments' => body['arguments'] || []) if body.key?('arguments')
+            opts = opts.merge('arguments' => args) if args.any?
```

Comment:
For both of these we should check that the user passed in a hash before calling `empty?` and raise an error if the value isn't a hash (I'm not sure if this gets called before or after we evaluate Puppet code, but using variables should also work and error if the variable evaluates to a non-hash).